src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -495,14 +509,68 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 		return tab.container ? Object.keys(tab.container)[0] : '';
 	}
 
-	private addNewTab(tab: TabConfig): void {
-		const existedTab = this.tabs.find(i => i.id === tab.id);
-		if (!existedTab) {
-			if (!tab.iconClass && tab.type !== 'group-header') {
-				tab.iconClass = 'default-tab-icon';
+	private async addNewTab(tab: TabConfig): Promise<void> {
+
+		try {
+			// Finding the owning extension for the tab
+			const owningExtension = this.extensionService.extensions.find(extension => {
+				if (extension?.contributes !== undefined && extension.contributes['dashboard.tabs']) {
+					const tabIndex = extension.contributes['dashboard.tabs'].findIndex(eTab => eTab.id === tab.id);
+					if (tabIndex !== -1) {
+						return true;
+					}
+				}
+				return false;
+			});
+
+			// If the tab is contributed by an extension, we'll wait for the extension to be activated before adding the tab.
+			// Not doing so causes issues with the tab UI not being rendered correctly.
+			if (owningExtension) {
+				// wait for the extension to be activated
+				await new Promise<void>((resolve, reject) => {
+					// If the extension is already activated, we can resolve immediately
+					if (this.extensionService.getExtensionsStatus()[owningExtension.id]?.activationTimes?.activateResolvedTime !== undefined) {
```

Comment:
Is there a reason you're not using something like https://github.com/microsoft/azuredatastudio/blob/25cdb558a707b55ada24b58489b9ad387dbde42d/src/vs/workbench/services/extensions/common/extensionHostManager.ts#L48? The way you're doing it could result in a race condition if the extension status is changed between when you check here and when the event is hooked up.

That being said - we already have the [onDashboardOpen](https://github.com/microsoft/azuredatastudio/blob/25cdb558a707b55ada24b58489b9ad387dbde42d/src/sql/workbench/api/browser/mainThreadDashboard.ts#L23) event which should cover this scenario too. Was the thought here not to block the entire dashboard just because we're waiting on the extensions to load?

Reply:
The extensions are already in the 'activating' state but are not done activating. This is the reason they cannot register the modelview for the dashboard tab and we see the blank screen.

Reply:
What you currently have has a race condition so probably shouldn't be the route we go with.

Activate will just be a no-op if it's already activated, so using that should be fine and avoids any similar problems (plus has the advantage of being a lot cleaner)

Reply:
Chatted offline - turns out there isn't anything available on any services to access this directly. So instead of modifying VS Code we're just going to fix the race condition.

It's possible there's a better way to do this, but I couldn't find a way immediately. And given the (hopefully) one-off nature of this change it should be fine.